### PR TITLE
Switch to constructor injection

### LINKED ISSUE #144: Switch to constructor inject instead of field injection

### PR DESCRIPTION
**Pull request by dacrome**
Resolves #144

### CODE REVIEW

File: src/main/java/org/osiam/Osiam.java [hunk 1]
```diff
@@ -2,20 +2,16 @@
 
 import com.google.common.collect.ImmutableMap;
 import com.ryantenney.metrics.spring.config.annotation.EnableMetrics;
-import com.zaxxer.hikari.HikariConfig;
-import com.zaxxer.hikari.HikariDataSource;
+import com.zaxxer.hikari.*;
```
Comment (wallner) [inline src/main/java/org/osiam/Osiam.java]: I'm not fundamentally opposed to wildcard  imports, even though I'm not a big fan of them. But please don't commit files with this PR that don't have any changes other than import collapses.

File: src/main/java/org/osiam/Osiam.java [hunk 2]
```diff
@@ -2,20 +2,16 @@
...
 import org.flywaydb.core.Flyway;
 import org.flywaydb.core.api.MigrationVersion;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.annotation.*;
```
Comment (wallner) [inline src/main/java/org/osiam/Osiam.java]: Please see above.

File: src/main/java/org/osiam/Osiam.java [hunk 3]
```diff
@@ -2,20 +2,16 @@
...
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.builder.SpringApplicationBuilder;
 import org.springframework.boot.context.web.SpringBootServletInitializer;
 import org.springframework.context.MessageSource;
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.EnableAspectJAutoProxy;
-import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.*;
```
Comment (wallner) [inline src/main/java/org/osiam/Osiam.java]: Please see comment above.

File: src/main/java/org/osiam/OsiamHome.java [hunk 4]
```diff
@@ -1,10 +1,8 @@
 package org.osiam;
 
 import com.google.common.base.Strings;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
+import org.slf4j.*;
```
Comment (wallner) [inline src/main/java/org/osiam/OsiamHome.java]: My comment about wildcard imports from the file `Osiam.java` holds true here as well.

**Comment by wallner** [comment]
I'm happy to merge this, but please take the two files out that only collapse imports.

File: src/main/java/org/osiam/Osiam.java (hunk 1 above)
Comment (dacrome) [inline src/main/java/org/osiam/Osiam.java]: I don't know why I changed the IntelliJ settings yesterday to use wildcard imports, but I'm also not a fan. I would like to introduce this as a default coding style for all osiam repositories.

Comment (dacrome) [inline src/main/java/org/osiam/Osiam.java]: So I meant to introduce single class import as [coding convention](https://github.com/osiam/osiam/blob/master/CONTRIBUTING.md#coding-conventions).

**Comment by wallner** [comment]
Thank you!

**Pull request merged**